behaviour/graql/reasoner/recursion.feature

```diff
@@ -0,0 +1,2025 @@
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+
+      $a isa person, has name "a";
+      $b isa person, has name "b";
+      $c isa person, has name "c";
+      $d isa person, has name "d";
+      $e isa person, has name "e";
+      $f isa person, has name "f";
+      $g isa person, has name "g";
+      $h isa person, has name "h";
+      $i isa person, has name "i";
+      $j isa person, has name "j";
+      $k isa person, has name "k";
+      $l isa person, has name "l";
+      $m isa person, has name "m";
+      $n isa person, has name "n";
+      $o isa person, has name "o";
+      $p isa person, has name "p";
+
+      (up-from: $a, up-to: $e) isa up;
+      (up-from: $a, up-to: $f) isa up;
+      (up-from: $f, up-to: $m) isa up;
+      (up-from: $g, up-to: $n) isa up;
+      (up-from: $h, up-to: $n) isa up;
+      (up-from: $i, up-to: $o) isa up;
+      (up-from: $j, up-to: $o) isa up;
+
+      (flat-from: $g, flat-to: $f) isa flat;
+      (flat-from: $m, flat-to: $n) isa flat;
+      (flat-from: $m, flat-to: $o) isa flat;
+      (flat-from: $p, flat-to: $m) isa flat;
+
+      (down-from: $l, down-to: $f) isa down;
+      (down-from: $m, down-to: $f) isa down;
+      (down-from: $g, down-to: $b) isa down;
+      (down-from: $h, down-to: $c) isa down;
+      (down-from: $i, down-to: $d) isa down;
+      (down-from: $p, down-to: $k) isa down;
+      """
+#    When materialised keyspace is completed
+    Then for graql query
+      """
+      match
+        (RSG-from: $x, RSG-to: $y) isa RevSG;
+        $x has name 'a';
+      get $y;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 3
+    Then answer set is equivalent for graql query
+      """
+      match
+        $y isa person, has name $name;
+        {$name == 'b';} or {$name == 'c';} or {$name == 'd';};
+      get $y;
+      """
+    Then for graql query
+      """
+      match
+        (RSG-from: $x, RSG-to: $y) isa RevSG;
+      get;
+      """
+#    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 11
+    Then answer set is equivalent for graql query
+      """
+      match
+        $x has name $nameX;
+        $y has name $nameY;
+        {$nameX == 'a';$nameY == 'b';} or {$nameX == 'a';$nameY == 'c';} or
+        {$nameX == 'a';$nameY == 'd';} or {$nameX == 'm';$nameY == 'n';} or
+        {$nameX == 'm';$nameY == 'o';} or {$nameX == 'p';$nameY == 'm';} or
+        {$nameX == 'g';$nameY == 'f';} or {$nameX == 'h';$nameY == 'f';} or
+        {$nameX == 'i';$nameY == 'f';} or {$nameX == 'j';$nameY == 'f';} or
+        {$nameX == 'f';$nameY == 'k';};
+      get $x, $y;
+      """
+#    Then materialised and reasoned keyspaces are the same size
+
+
+  # TODO: re-enable all steps when resolvable (currently takes too long) (#75)
+  Scenario: dual linear transitivity matrix test
+
+    test 6.1 from Cao p 71
```

Comment:
Just to give the full reference. It's Cao - Methods for evaluating queries toHorn knowledge bases in first-order logic

Reply:
Citation added